.github/workflows/integration-tests.yml

```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - master
+  pull_request:
```

Comment:
There's a tricky balance here — allowing integration tests in PRs will add a risk to spending the OpenAI budget by everyone, basically. Let's remove this for now